docs/tutorials/using_apis/Mbed_USB_WAV_Audio_Player/Mbed_USB_WAV_Audio_Player.md

```diff
@@ -0,0 +1,88 @@
+
+### What You Need
+- Mbed Enabled board
+- SD card
+- SD card module (if board does not already have one)
+- USB cable
+
+### Setup
+The following steps demonstrate the setup and use of the Mbed WAV Audio Player:
+1. Create a new Mbed project.   
+`mbed new <project name>`
+2. Add the [sd-driver](https://github.com/ARMmbed/sd-driver) and [AudioPlayer](https://github.com/c1728p9/AudioPlayer) libraries into the new Mbed project directory.   
+`mbed add https://github.com/ARMmbed/sd-driver`    
+`mbed add https://github.com/c1728p9/AudioPlayer`
+3. Copy and paste the below example code into *main.cpp*.
+4. Load WAV file(s) onto SD card. The example below uses a public domain WAV file called "Bach-minuet-in-g.wav" (attached below for download) that is inside of a "songs" directory on the SD card. It is important that the WAV file be PCM signed 16-bit little endian, or else it will not play. The WAV file can have any sampling rate and can have any number of channels. [Online-convert](https://audio.online-convert.com/convert-to-wav) can be used to achieve WAV files with different formats from any source audio file.   
+![N](https://raw.githubusercontent.com/mrcoulter45/mbed-os-5-docs/Mbed_USB_WAV_Audio_Player_Tutorial/docs/tutorials/using_apis/Mbed_USB_WAV_Audio_Player/images/Mbed_USB_WAV_Audio_Player_img1.png)
+5. Compile the program and flash the binary.
+6. Ensuring that the USBPhy is connected to the host computer, select "Mbed Audio" as the host PC's default audio input device. On Windows, in Control Panel > Hardware and Sound > Sound, in the recording tab, right click on "Microphone Mbed Audio" > Properties > Listen. Check the "Listen to this Device" box and click Apply. The audio from the WAV file should now be audible through whichever output device is selected for the host PC's audio output.   
+![N](https://raw.githubusercontent.com/mrcoulter45/mbed-os-5-docs/Mbed_USB_WAV_Audio_Player_Tutorial/docs/tutorials/using_apis/Mbed_USB_WAV_Audio_Player/images/Mbed_USB_WAV_Audio_Player_img2.PNG)
+![N](https://raw.githubusercontent.com/mrcoulter45/mbed-os-5-docs/Mbed_USB_WAV_Audio_Player_Tutorial/docs/tutorials/using_apis/Mbed_USB_WAV_Audio_Player/images/Mbed_USB_WAV_Audio_Player_img3.PNG)
+7. See "Troubleshooting" if issues persist.
+ 
+### main.cpp
+```c++
+// Mbed WAV Audio Player
+#include "mbed.h"
+#include "USBAudio.h"
+#include "SDBlockDevice.h"
+#include "FATFileSystem.h"
+#include "AudioPlayer.h"
+#include "WaveAudioStream.h"
+
+// Connection for SD card
+SDBlockDevice sd(PTE3, PTE1, PTE2, PTE4);//MOSI, MISO, SCLK, CS
+FATFileSystem fs("sd", &sd);
+
+int main() {
+    // Set the maximum speed so it can keep up with audio
+    sd.frequency(25000000);
+    // Load WAV file from SD card
+    // WAV file must be PCM signed 16-bit little endian
+    File file(&fs, "songs/Bach-minuet-in-g.wav");
```

Comment:
It may be good to use the `open` call rather than the constructor to open the file. This would allow return code to be checked. Something like:
```C++
    if (file.open(&fs, "songs/Bach-minuet-in-g.wav") != 0) {
        printf("Could not open 'songs/Bach-minuet-in-g.wav'\r\n");
        return;
    }
```